src/stripeDotnetLanguageServer/stripe.LanguageServer/WorkspaceManager.cs

```diff
@@ -0,0 +1,87 @@
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.MSBuild;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.Extensions.Logging;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using System.IO;
+using System.Linq;
+
+namespace stripe.LanguageServer
+{
+    public class WorkspaceManager
+    {
+        private readonly ILogger<WorkspaceManager> _logger;
+
+        // Workspace for project context
+        private Workspace _workspace;
+
+        private Solution _currentSolution;
+
+        public WorkspaceManager(ILogger<WorkspaceManager> logger, Workspace workspace)
+        {
+            _logger = logger;
+            _workspace = workspace;
+            _currentSolution = workspace.CurrentSolution;
+        }
+
+        public Workspace GetWorkspace()
+        {
+            return _workspace;
+        }
+
+        public Solution GetCurrentSolution()
+        {
+            return _currentSolution;
+        }
+
+        public Document GetDocument(DocumentUri uri)
+        {
+            var documentId = this._currentSolution.GetDocumentIdsWithFilePath(uri.GetFileSystemPath()).FirstOrDefault();
+            _logger.LogDebug("Getting DocumentId: " + documentId);
+            return this._currentSolution.GetDocument(documentId);
+        }
+
+        // Handles OpenDocument events
+        // If document already exists in the project, return it. If document does not exist, it can be for two reasons:
+        //  1. the .cs file is a metdata file that was opened. i.e. a library reference. In this case, the scheme of the uri returned will be "omnisharp-metadata"
+        //  2. the .cs file is a new file created for the project.
+        public void OpenDocument(DocumentUri uri)
```

Comment:
Nit: These are already readable, but I feel it could be more readable if they were called `HandleDidOpenTextDocument` and `HandleDidUpdateTextDocument`. But I don't know anything about C# conventions so this is perfectly fine to me